File: Python/examples/example_linear_gain_read.py
# Example: set gain if LINEAR, then read snapshot in mV or W
# Requirements: pip install pyserial

PORT = "COM5"  # change to your COM port
READ_MODE = "mV"  # "mV" or "W"
HEAD = 1
GAIN = 0

import os, sys
sys.path.append(os.path.dirname(os.path.dirname(__file__)))
from coredaq_python_api import CoreDAQ
dev = CoreDAQ(PORT)

dev.soft_zero_from_snapshot()

try:
    if dev.frontend_type() == CoreDAQ.FRONTEND_LINEAR:
        dev.set_gain(HEAD, GAIN)
        print(f"Set gain: head={HEAD} gain={GAIN}")

    if READ_MODE.upper() == "W":
        watts = dev.snapshot_W(n_frames=1)
        print("W:", watts)
    else:
        mv, gains = dev.snapshot_mV(n_frames=1)
        print("mV:", mv)
        if dev.frontend_type() == CoreDAQ.FRONTEND_LINEAR:
            print("Gains:", gains)
finally:
    dev.close()

File: JavaScript/examples/example_linear_gain_read.js
// Example: set gain if LINEAR, then read snapshot in mV or W
// Requirements: npm install serialport

const PORT = 'COM3'; // change to your COM port
const READ_MODE = 'mV'; // 'mV' or 'W'
const HEAD = 1;
const GAIN = 3;

const { CoreDAQ } = require('../coredaq_js_api');

(async () => {
  const dev = await CoreDAQ.open(PORT);
  try {
    if (dev.frontend_type() === CoreDAQ.FRONTEND_LINEAR) {
      await dev.set_gain(HEAD, GAIN);
      console.log(`Set gain: head=${HEAD} gain=${GAIN}`);
    }

    if (READ_MODE.toUpperCase() === 'W') {
      const watts = await dev.snapshot_W(1);
      console.log('W:', watts);
    } else {
      const [mv, gains] = await dev.snapshot_mV(1);
      console.log('mV:', mv);
      if (dev.frontend_type() === CoreDAQ.FRONTEND_LINEAR) {
        console.log('Gains:', gains);
      }
    }
  } finally {
    await dev.close();
  }
})();